python/tk_houdini/ui_generation.py

```diff
@@ -882,6 +913,66 @@ def _format_xml(xml):
 
     return formatted_xml
 
+def _on_file_change_timeout():
+    """
+    Checks to see if the current file has changed. If it has, try to set the
+    new context for the file.
+    """
+
+    import hou
+    cur_file = hou.hipFile.path()
+
+    global g_current_file
+    if cur_file == g_current_file:
+        # the current file is the same as it was last time. no file change,
+        # no need to proceed
+        return
+
+    # update the current file global so that the next timeout won't do anything
+    # it isn't supposed to
+    g_current_file = cur_file
+
+    import sgtk
+    cur_engine = None
+
+    # attempt to get the current engine and context
+    try:
+        cur_engine = sgtk.platform.current_engine()
+        cur_context = cur_engine.context
+        engine_name = cur_engine.name
+    except Exception, e:
+        engine_name = "tk-houdini"
+        cur_context = None
+
+    try:
+        tk = sgtk.tank_from_path(cur_file)
+    except sgtk.TankError, e:
+        # Unable to get tk api instance from the path. won't be able to get a
+        # new context. if there is an engine running, destroy it.
+        if cur_engine:
+            cur_engine.destroy()
+        return
+
+    new_context = tk.context_from_path(cur_file, cur_context)
+
+    if cur_context == new_context:
+        return
+
+    if cur_engine:
+        cur_engine.destroy()
+
+    # try to create new engine
+    try:
+        sgtk.platform.start_engine(engine_name, tk, new_context)
```

Comment:
Should this become switch context to take advantage of the new wiz bang quick context switching?